### DISCUSSION TITLE
Clicking "Sign in with credentials" on localhost:3000 redirects to myproject.vercel.app/api/auth/signin?csrf=true

### DISCUSSION THREAD

**Discussion by Glutch** [nextauthjs/next-auth#1318] [Help]
Clicking "Sign in with credentials" on localhost:3000 redirects to myproject.vercel.app/api/auth/signin?csrf=true

This results in that i cannot reach the admin page on localhost. I'm not sure how this is even happening

[...nextauth].js

```
import NextAuth from "next-auth"
import Providers from "next-auth/providers"

const isCorrectCredentials = credentials =>
  credentials.username === 'awdawdwad' &&
  credentials.password === 'awda'

const options = {
  providers: [
    Providers.Credentials({
      name: "Credentials",
      credentials: {
        username: { label: "Username", type: "text", placeholder: "username" },
        password: { label: "Password", type: "password" },
      },
      authorize: async credentials => {
        if (isCorrectCredentials(credentials)) {
          const user = { id: 1, name: "Admin" }
          return Promise.resolve(user)
        } else {
          return Promise.resolve(null)
        }
      },
    }),
  ],
}

export default (req, res) => NextAuth(req, res, options)
```

**Comment by oliver-daniel** [comment]
Do you have `NEXTAUTH_URL` set in a `.env` file?

**Comment by Glutch** [comment]
> Do you have `NEXTAUTH_URL` set in a `.env` file?

Yes, i removed it and it now works fine on localhost. Thanks

But in production (with NEXTAUTH_URL set to my https://domainname.com) the behaviour is really strange
1. go to https://domainname.com/admin
2. youre not allowed to be here [sign in] *click*
3. see sign in with credentials page (enter username & password, or literally anything, doesnt matter. click sign in)
4. gets redirected to the signin page on https://domainname-n8nokkeo.vercel.app/ (the latest commit from vercel)
5. get signed in, but i land on root at https://domainname-n8nokkeo.vercel.app/ instead of https://domainname.com/admin
